File: sdk/nodejs/src/resources/vcloudstack.js
"use strict";

const { BaseResource } = require("./base");

class VCloudStackResource extends BaseResource {
  async rescue(serviceId, vmId) {
    return this._http.post("/api/service/{id}/vms/{vmid}/rescue");
  }

  async unrescue(serviceId, vmId) {
    return this._http.post("/api/service/{id}/vms/{vmid}/unrescue");
  }

  async getConsole(serviceId, vmId) {
    return this._http.get("/api/service/{id}/vms/{vmid}/console");
  }

  async getUsage(serviceId, vmId) {
    return this._http.get("/api/service/{id}/vms/{vmid}/usage");
  }

}

module.exports = { VCloudStackResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Vcloudstack.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Vcloudstack extends BaseResource {

    public Vcloudstack(HttpClient http) {
        super(http);
    }

    /** Enter rescue */
    public Map<String, Object> rescue(int id, int vmid) throws Exception {
        return http.post(String.format("/api/service/%id/vms/%vmid/rescue", id, vmid), null);
    }

    /** Exit rescue */
    public Map<String, Object> unrescue(int id, int vmid) throws Exception {
        return http.post(String.format("/api/service/%id/vms/%vmid/unrescue", id, vmid), null);
    }

    /** Get console URL */
    public Map<String, Object> console(int id, int vmid) throws Exception {
        return http.get(String.format("/api/service/%id/vms/%vmid/console", id, vmid));
    }

    /** Get usage */
    public Map<String, Object> usage(int id, int vmid) throws Exception {
        return http.get(String.format("/api/service/%id/vms/%vmid/usage", id, vmid));
    }

}
